doc/introduction/measurements.rst

```diff
@@ -524,6 +524,87 @@ Collecting statistics for sequences of mid-circuit measurements is supported wit
+  with finite-shot circuits. Use ``postselect_mode="hw-like"`` to discard invalid samples. In this case, the number
+  of samples that are used for processing results can be less than the total number of shots. If
+  ``postselect_mode="fill-shots"`` is used, then the postselected value will be sampled unconditionally, and all
+  samples will be valid. This is equivalent to sampling until the number of valid samples matches the total number
+  of shots. The default behaviour is ``postselect_mode="hw-like"``.
+
+  .. code-block:: python3
+
+      import pennylane as qml
+      import numpy as np
+
+      dev = qml.device("default.qubit", wires=3, shots=10)
+
+      def circuit(x):
+          qml.RX(x, 0)
+          m0 = qml.measure(0, postselect=1)
+          qml.CNOT([0, 1])
+          return qml.sample(qml.PauliZ(0))
+
+      fill_shots_qnode = qml.QNode(circuit, dev, mcm_method="one-shot", postselect_mode="fill-shots")
+      hw_like_qnode = qml.QNode(circuit, dev, mcm_method="one-shot", postselect_mode="hw-like")
+
+  >>> fill_shots_qnode(np.pi / 2)
+  array([-1., -1., -1., -1., -1., -1., -1., -1., -1., -1.])
+  >>> hw_like_qnode(np.pi / 2)
+  array([-1., -1., -1., -1., -1., -1., -1.])
+
+  .. note::
+
+      When using the ``jax`` interface, ``postselect_mode="hw-like"`` will have different behaviour based on the
+      chosen ``mcm_method``.
+
+      * If ``mcm_method="one-shot"``, invalid shots will not be discarded. Instead, invalid samples will be replaced
+        by ``np.iinfo(np.int32).min``. These invalid samples will not be used for processing final results (like
+        expectation values), but will appear in the ``QNode`` output if samples are requested directly. Consider
+        the circuit below:
+
+        .. code-block:: python3
+
+            import pennylane as qml
+            import jax
+            import jax.numpy as jnp
+
+            dev = qml.device("default.qubit", wires=3, shots=10, seed=jax.random.PRNGKey(123))
+
+            @qml.qnode(dev, postselect_mode="hw-like", mcm_method="one-shot")
+            def circuit(x):
+                qml.RX(x, 0)
+                qml.measure(0, postselect=1)
+                return qml.sample(qml.PauliZ(0))
+
+        >>> x = jnp.array(1.8)
+        >>> f(x)
+        Array([-2.1474836e+09, -1.0000000e+00, -2.1474836e+09, -2.1474836e+09,
+               -1.0000000e+00, -2.1474836e+09, -1.0000000e+00, -2.1474836e+09,
+               -1.0000000e+00, -1.0000000e+00], dtype=float32, weak_type=True)
+
+      * When using ``jax.jit``, using ``mcm_method="deferred"`` is not supported with ``postselect_mode="hw-like"``.
+        Therefore, the default behaviour will be to use ``postselect_mode="fill-shots"``. This is due to limitations
+        of the :func:`~pennylane.defer_measurements` transform, and this behaviour will change in the future to be
+        more consistent with ``mcm_method="one-shot"``.
+
```

Comment:
Nice! Although this is confusing, I think it's just the nature of the situation and you've explained it as best as we can for now.